src/python/turicreate/toolkits/image_analysis/image_analysis.py

```diff
@@ -192,6 +192,171 @@ def resize(image, width, height, channels=None, decode=False, resample="nearest"
+
+    Parameters
+    ----------
+
+    image : turicreate.Image | SArray
+        The image or SArray of images to be resized.
+    min_side_length : int
+        The size of the minimum side of the image to be resized while maintaining aspect ratio.
+        For ex: 
+            if the image shape is : 720 * 960 i.e => the aspect ratio is 3:4
+            so if you set the min_side_length to 600, then the resized image dimension would be => 600 * 800
+    max_side_length : int
+        The size of the maximum side of the image to be resized while maintaining aspect ratio.
+        For ex: 
+            if the image shape is : 720 * 960 i.e => the aspect ratio is 3:4
+            so if you set the max_side_length to 600, then the resized image dimension would be => 450 * 600
+    channels : int, optional
+        The number of channels the image is resized to. 1 channel
+        corresponds to grayscale, 3 channels corresponds to RGB, and 4
+        channels corresponds to RGBA images.
+    decode : bool, optional
+        Whether to store the resized image in decoded format. Decoded takes
+        more space, but makes the resize and future operations on the image faster.
+    resample : 'nearest' or 'bilinear'
+        Specify the resampling filter:
+
+            - ``'nearest'``: Nearest neigbhor, extremely fast
+            - ``'bilinear'``: Bilinear, fast and with less aliasing artifacts
+
+    Returns
+    -------
+    out : turicreate.Image
+        Returns a resized Image object with original aspect ratio.
+
+    Notes
+    -----
+    Grayscale Images -> Images with one channel, representing a scale from
+    white to black
+
+    RGB Images -> Images with 3 channels, with each pixel having Green, Red,
+    and Blue values.
+
+    RGBA Images -> An RGB image with an opacity channel.
+
+    Examples
+    --------
+
+    Resize a single image
+
+    >>> img = turicreate.Image('https://static.turi.com/datasets/images/sample.jpg')
+    >>> resized_img = turicreate.image_analysis.resize_with_original_aspect_ratio(img,min_side_length=500,max_side_length=None, channels=1)
+    >>> resized_img = turicreate.image_analysis.resize_with_original_aspect_ratio(img,min_side_length=None,max_side_length=500, channels=1)
+
+    Resize an SArray of images
+
+    >>> url ='https://static.turi.com/datasets/images/nested'
+    >>> image_sframe = turicreate.image_analysis.load_images(url, "auto", with_path=False,
+    ...                                                    recursive=True)
+    >>> image_sarray = image_sframe["image"]
+    >>> resized_images = turicreate.image_analysis.resize(image_sarray, min_side_length=500,max_side_length=None, channels=1)
+    >>> resized_images = turicreate.image_analysis.resize(image_sarray, min_side_length=None,max_side_length=500, channels=1)
+    """
+
+    from ...data_structures.sarray import SArray as _SArray
+    import turicreate as _tc
```

Comment:
You don't need this line. It's already imported at the top of the file.

Reply:
Will remove it from there, also I want to point out that the same thing is present in the `resize` function.